providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/resolver/process/storage/connector/grpc/GrpcStreamConnector.java

```diff
@@ -0,0 +1,129 @@
+package dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.grpc;
+
+import dev.openfeature.contrib.providers.flagd.FlagdOptions;
+import dev.openfeature.contrib.providers.flagd.resolver.common.ChannelBuilder;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.Connector;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.StreamPayload;
+import dev.openfeature.contrib.providers.flagd.resolver.process.storage.connector.StreamPayloadType;
+import io.grpc.ManagedChannel;
+import lombok.extern.java.Log;
+import sync.v1.FlagSyncServiceGrpc;
+import sync.v1.SyncService;
+
+import java.util.Random;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.logging.Level;
+
+/**
+ * Implements the {@link Connector} contract and emit flags obtained from flagd sync gRPC contract.
+ */
+@Log
+public class GrpcStreamConnector implements Connector {
+    private static final Random RANDOM = new Random();
+
+    private static final int INIT_BACK_OFF = 2 * 1000;
+    private static final int MAX_BACK_OFF = 120 * 1000;
+
+    private final AtomicBoolean shutdown = new AtomicBoolean(false);
+    private final BlockingQueue<StreamPayload> blockingQueue = new LinkedBlockingQueue<>(5);
```

Comment:
I wonder what could be a good experiment to run for identifying the right magic number for the queue capacity 🤔 do you have any idea? @toddbaert  @Kavindu-Dodan

Reply:
I don't think we need to worry much about this. Readers are all about parsing payload and writing flags to a map. This is fast enough compared to gRPC stream coming through wire at write end.